lib/feature/main/main_screen.dart

```diff
@@ -1,22 +1,88 @@
 import 'package:flutter/cupertino.dart';
 import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
 import 'package:netguru_flutter_template/app/s.dart';
+import 'package:netguru_flutter_template/feature/main/main_cubit.dart';
+import 'package:netguru_flutter_template/injection/injection.dart';
+import 'package:netguru_flutter_template/widget/custom_loading/custom_loading.dart';
+import 'package:netguru_flutter_template/values/app_theme.dart';
 
-class MainScreen extends StatelessWidget {
+class MainScreenRoute extends MaterialPageRoute<bool> {
+  MainScreenRoute() : super(builder: (_) => MainScreen());
+}
+
+class MainScreen extends StatefulWidget {
   @override
-  Widget build(BuildContext context) => Scaffold(
-        appBar: AppBar(title: Text(S.of(context).appName)),
-        body: Center(
-          child: Column(
-            mainAxisSize: MainAxisSize.min,
-            children: [
-              Text('Hello World!'),
-            ],
-          ),
-        ),
-      );
+  _MainScreenState createState() => _MainScreenState();
 }
 
-class MainScreenRoute extends MaterialPageRoute<bool> {
-  MainScreenRoute() : super(builder: (_) => MainScreen());
+class _MainScreenState extends State<MainScreen> {
+  MainCubit? mainCubit;
+
+  bool isLoading = false;
+
+  @override
+  void initState() {
+    mainCubit = getIt.get<MainCubit>();
+    mainCubit?.fetchListItems();
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider<MainCubit>(
+      create: (context) => mainCubit!,
+      child: BlocConsumer<MainCubit, MainState>(
+        listener: (BuildContext context, state) {
```

Comment:
so in our project here there is nothing to listen to, so there is no need for the listener,  and this will let us realize that BlocBuilder would be enough instead of Bloc Consumer